LVGL does not build for ESP32-S2

We use GitHub issues for development related discussions.
Please use the [forum](https://forum.littlevgl.com/) to ask questions.

**Describe the issue**

When compiling the build fails: (see below).

I made a fix at https://github.com/cmumford/lv_port_esp32/commit/8697e2cdd05ba3d49393a52f43959cdb92c178a9. However, I don't yet have an attached screen, so it is untested - it only compiles. I will create a PR for that commit when I have a chance to test the change. (maybe this weekend :crossed_fingers:)

**Code to reproduce the issue**

Follow getting started guide: https://github.com/lvgl/lv_port_esp32#get-started except specify the esp32s2 target:

```sh
idf.py set-target esp32s2
idf.py menuconfig
```

**Expected Results**

Successful build

**Actual Results**

Failure as shown below

**ESP32 Chip version**

ESP32-S2

**ESP-IDF version**

ESP-IDF v4.3-dev-1901-g178b122c1

**Development kit used**

Cucumber R ESP32-S2 (but never ran on it)

**Development machine OS**

macOS: 10.15.7

**Compilation warnings/errors (if available)**

```
../components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_helpers.c: In function 'lvgl_spi_driver_init':
../components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_helpers.c:204:35: error: 'VSPI_HOST' undeclared (first use in this function); did you mean 'FSPI_HOST'?
     assert((SPI_HOST <= host) && (VSPI_HOST >= host));
                                   ^~~~~~~~~
~/esp/esp-idf/components/esp_common/include/esp_compiler.h:29:25: note: in definition of macro 'likely'
 #define likely(x)      (x)
                         ^
../components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_helpers.c:204:5: note: in expansion of macro 'assert'
     assert((SPI_HOST <= host) && (VSPI_HOST >= host));
     ^~~~~~
../components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_helpers.c:204:35: note: each undeclared identifier is reported only once for each function it appears in
     assert((SPI_HOST <= host) && (VSPI_HOST >= host));
                                   ^~~~~~~~~
~/esp/esp-idf/components/esp_common/include/esp_compiler.h:29:25: note: in definition of macro 'likely'
 #define likely(x)      (x)
                         ^
../components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_helpers.c:204:5: note: in expansion of macro 'assert'
     assert((SPI_HOST <= host) && (VSPI_HOST >= host));
```

--- a/components/lvgl_esp32_drivers/lvgl_helpers.c
+++ b/components/lvgl_esp32_drivers/lvgl_helpers.c
@@ -200,11 +200,17 @@ bool lvgl_spi_driver_init(int host,
     int dma_channel,
     int quadwp_pin, int quadhd_pin)
 {
+#if defined (CONFIG_IDF_TARGET_ESP32)
     assert((SPI_HOST <= host) && (VSPI_HOST >= host));
-
     const char *spi_names[] = {
         "SPI_HOST", "HSPI_HOST", "VSPI_HOST"
     };
+#elif defined (CONFIG_IDF_TARGET_ESP32S2)
+    assert((SPI_HOST <= host) && (HSPI_HOST >= host));
+    const char *spi_names[] = {
+        "SPI_HOST", "", ""
+    };
+#endif
 
     ESP_LOGI(TAG, "Configuring SPI host %s (%d)", spi_names[host], host);
     ESP_LOGI(TAG, "MISO pin: %d, MOSI pin: %d, SCLK pin: %d, IO2/WP pin: %d, IO3/HD pin: %d",

--- a/components/lvgl_esp32_drivers/lvgl_spi_conf.h
+++ b/components/lvgl_esp32_drivers/lvgl_spi_conf.h
@@ -64,11 +64,13 @@ extern "C" {
 
 #define ENABLE_TOUCH_INPUT  CONFIG_LV_ENABLE_TOUCH
 
-#if CONFIG_LV_TFT_DISPLAY_SPI_HSPI == 1
+#if defined (CONFIG_LV_TFT_DISPLAY_SPI_HSPI)
 #define TFT_SPI_HOST HSPI_HOST
-#else
+#elif defined (CONFIG_LV_TFT_DISPLAY_SPI_VSPI)
 #define TFT_SPI_HOST VSPI_HOST
-#endif /*CONFIG_LV_TFT_DISPLAY_SPI_HSPI == 1*/
+#elif defined (CONFIG_LV_TFT_DISPLAY_SPI_FSPI)
+#define TFT_SPI_HOST FSPI_HOST
+#endif
 
 #if defined (CONFIG_LV_TFT_DISPLAY_SPI_HALF_DUPLEX)
 #define DISP_SPI_HALF_DUPLEX
@@ -88,7 +90,9 @@ extern "C" {
 #define TOUCH_SPI_HOST HSPI_HOST
 #elif defined (CONFIG_LV_TOUCH_CONTROLLER_SPI_VSPI)
 #define TOUCH_SPI_HOST VSPI_HOST
-#endif /*CONFIG_LV_TOUCH_CONTROLLER_SPI_HSPI == 1*/
+#elif defined (CONFIG_LV_TOUCH_CONTROLLER_SPI_FSPI)
+#define TOUCH_SPI_HOST FSPI_HOST
+#endif
 
 /* Handle the FT81X Special case */
 #if defined (CONFIG_LV_TFT_DISPLAY_CONTROLLER_FT81X)
@@ -103,7 +107,7 @@ extern "C" {
 // Detect the use of a shared SPI Bus and verify the user specified the same SPI bus for both touch and tft
 #if defined (CONFIG_LV_TOUCH_DRIVER_PROTOCOL_SPI) && TP_SPI_MOSI == DISP_SPI_MOSI && TP_SPI_CLK == DISP_SPI_CLK
 #if TFT_SPI_HOST != TOUCH_SPI_HOST
-#error You must specify the same SPI host (HSPI or VSPI) for both display and touch driver
+#error You must specify the same SPI host (HSPI, VSPI or FSPI) for both display and touch driver
 #endif
 
 #define SHARED_SPI_BUS

--- a/components/lvgl_esp32_drivers/lvgl_tft/Kconfig
+++ b/components/lvgl_esp32_drivers/lvgl_tft/Kconfig
@@ -401,7 +401,8 @@ menu "LVGL TFT Display controller"
     choice
         prompt "TFT SPI Bus." if LV_TFT_DISPLAY_PROTOCOL_SPI && \
             !LV_PREDEFINED_DISPLAY_TTGO
-        default LV_TFT_DISPLAY_SPI_VSPI if LV_PREDEFINED_DISPLAY_TTGO
+        default LV_TFT_DISPLAY_SPI_VSPI if LV_PREDEFINED_DISPLAY_TTGO && \
+            !IDF_TARGET_ESP32S2
         default LV_TFT_DISPLAY_SPI_HSPI
         help
               Select the SPI Bus the TFT Display is attached to.
@@ -409,7 +410,9 @@ menu "LVGL TFT Display controller"
         config LV_TFT_DISPLAY_SPI_HSPI
             bool "HSPI"
         config LV_TFT_DISPLAY_SPI_VSPI
-            bool "VSPI"
+            bool "VSPI" if !IDF_TARGET_ESP32S2
+        config LV_TFT_DISPLAY_SPI_FSPI
+            bool "FSPI" if IDF_TARGET_ESP32S2
     endchoice
 
     choice

--- a/components/lvgl_esp32_drivers/lvgl_tft/Kconfig
+++ b/components/lvgl_esp32_drivers/lvgl_tft/Kconfig
@@ -403,7 +403,7 @@ menu "LVGL TFT Display controller"
             !LV_PREDEFINED_DISPLAY_TTGO
         default LV_TFT_DISPLAY_SPI_VSPI if LV_PREDEFINED_DISPLAY_TTGO && \
             !IDF_TARGET_ESP32S2
-        default LV_TFT_DISPLAY_SPI_HSPI
+        default LV_TFT_DISPLAY_SPI_FSPI if IDF_TARGET_ESP32S2
         help
               Select the SPI Bus the TFT Display is attached to.
 

--- a/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
+++ b/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
@@ -86,7 +86,7 @@ menu "LVGL Touch controller"
             bool "HSPI"
         config LV_TOUCH_CONTROLLER_SPI_VSPI
             bool "VSPI" if !IDF_TARGET_ESP32S2
-        config LV_TFT_DISPLAY_SPI_FSPI
+        config LV_TOUCH_DISPLAY_SPI_FSPI
             bool "FSPI" if IDF_TARGET_ESP32S2
         endchoice
     

--- a/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
+++ b/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
@@ -78,7 +78,7 @@ menu "LVGL Touch controller"
         depends on LV_TOUCH_DRIVER_PROTOCOL_SPI
         
         default LV_TOUCH_CONTROLLER_SPI_VSPI if !IDF_TARGET_ESP32S2
-        default LV_TOUCH_CONTROLLER_SPI_HSPI if IDF_TARGET_ESP32S2
+        default LV_TOUCH_CONTROLLER_SPI_FSPI if IDF_TARGET_ESP32S2
         help
             Select the SPI Bus the TFT Display is attached to.
     

--- a/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
+++ b/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
@@ -86,7 +86,7 @@ menu "LVGL Touch controller"
             bool "HSPI"
         config LV_TOUCH_CONTROLLER_SPI_VSPI
             bool "VSPI" if !IDF_TARGET_ESP32S2
-        config LV_TOUCH_DISPLAY_SPI_FSPI
+        config LV_TOUCH_CONTROLLER_SPI_FSPI
             bool "FSPI" if IDF_TARGET_ESP32S2
         endchoice
     

--- a/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
+++ b/components/lvgl_esp32_drivers/lvgl_touch/Kconfig
@@ -77,14 +77,17 @@ menu "LVGL Touch controller"
         prompt "Touch Controller SPI Bus."
         depends on LV_TOUCH_DRIVER_PROTOCOL_SPI
         
-        default LV_TOUCH_CONTROLLER_SPI_VSPI
+        default LV_TOUCH_CONTROLLER_SPI_VSPI if !IDF_TARGET_ESP32S2
+        default LV_TOUCH_CONTROLLER_SPI_HSPI if IDF_TARGET_ESP32S2
         help
             Select the SPI Bus the TFT Display is attached to.
     
         config LV_TOUCH_CONTROLLER_SPI_HSPI
             bool "HSPI"
         config LV_TOUCH_CONTROLLER_SPI_VSPI
-            bool "VSPI"
+            bool "VSPI" if !IDF_TARGET_ESP32S2
+        config LV_TFT_DISPLAY_SPI_FSPI
+            bool "FSPI" if IDF_TARGET_ESP32S2
         endchoice
     
     menu "Touchpanel (XPT2046) Pin Assignments"

